website/docs/developer/research/research-deep-glacier-migration.md

```diff
@@ -0,0 +1,174 @@
+
+- After objects are restored, we can use a lambda function or a script that will run after 12 hours and copy the restored objects to deep archive S3 storage using boto3 `copy_object` function as shown below. 
+
+:::important
+Note that there should be some confirmation that all requested objects are successfully restored before running this script.
+:::
+
+```python
+import boto3
+
+s3 = boto3.client('s3')
+#this list should be provided by the user
+retrieve_objects = ['test1.xml', 'test2.xml']
+bucket_name = 'orca-rhassan-sandbox-glacier'
+for migrate_object in retrieve_objects:
+    try:
+        response = s3.copy_object(
+            Bucket= bucket_name,  #bucket name where the object is stored currently
+            CopySource={'Bucket': bucket_name, 'Key': migrate_object},
+            Key = migrate_object, #keyname of object after migration
+            StorageClass='DEEP_ARCHIVE'
+        )
+        print(response)
+        print(f"copy succeeded for object {migrate_object} having E tag " + response["CopyObjectResult"]["ETag"])
+    except Exception as ex:
+        print(ex)
+```
+Output:
+```
+{'ResponseMetadata': {'RequestId': 'EM3XDZ6BB7FJJ88W', 'HostId': '/HayQvl8E2cCsclvO0b4MQUHAJnf7exmJhCILDsXfCieuaJIrAt5ZD9ocOkaam6rQbgXt0qx18A=', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amz-id-2': '/HayQvl8E2cCsclvO0b4MQUHAJnf7exmJhCILDsXfCieuaJIrAt5ZD9ocOkaam6rQbgXt0qx18A=', 'x-amz-request-id': 'EM3XDZ6BB7FJJ88W', 'date': 'Sun, 24 Jul 2022 18:13:32 GMT', 'x-amz-storage-class': 'DEEP_ARCHIVE', 'content-type': 'application/xml', 'server': 'AmazonS3', 'content-length': '234'}, 'RetryAttempts': 0}, 'CopyObjectResult': {'ETag': '"66d8119b127d165f8bc66fa66e0e35a6"', 'LastModified': datetime.datetime(2022, 7, 24, 18, 13, 32, tzinfo=tzutc())}}
+copy succeeded for object test1.xml having E tag "66d8119b127d165f8bc66fa66e0e35a6"
+{'ResponseMetadata': {'RequestId': 'SA312G5B2RRN5P0C', 'HostId': '/YIl4aUR+CAov4/pxzzd8oVORmYYsewTbwZfnVDkWRa+8g/epeQ7KEZGqSVgcx5LheMhU/dXMfw=', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amz-id-2': '/YIl4aUR+CAov4/pxzzd8oVORmYYsewTbwZfnVDkWRa+8g/epeQ7KEZGqSVgcx5LheMhU/dXMfw=', 'x-amz-request-id': 'SA312G5B2RRN5P0C', 'date': 'Sun, 24 Jul 2022 18:13:33 GMT', 'x-amz-storage-class': 'DEEP_ARCHIVE', 'content-type': 'application/xml', 'server': 'AmazonS3', 'content-length': '234'}, 'RetryAttempts': 0}, 'CopyObjectResult': {'ETag': '"14694ab9e3089a7c29e0dc9add4a82ab"', 'LastModified': datetime.datetime(2022, 7, 24, 18, 13, 33, tzinfo=tzutc())}}
+copy succeeded for object test2.xml having E tag "14694ab9e3089a7c29e0dc9add4a82ab"
+```
+- Once objects are copied successfully, the old objects in standard storage can be deleted using boto3 client `delete_object` API call per object or `delete_objects` which can delete upto 1000 objects with a single API call.
```

Comment:
Since they are copied into the same spot with a different storage type do they still need to be deleted? If they do need to be deleted, are there certain checks/safeguards that should be thought about and coded to make sure we do not accidentally remove something? What additional prorated charges can we expect to incur?

Reply:
yes they still need to be deleted. There is currently no move_object API in boto3 to directly do the migration. 
In the code, we should add some prefix to store the restored objects like `restored/`. 

Since objects restored will be in "standard" storage, we will be charged "$0.023 per GB for First 50 TB / Month" and "$0.023 per GB for Next 450 TB / Month". We should delete the old objects within a few days to avoid this charge

Reply:
objects restored will not incur a charge.